### PR TITLE
Add links to documentation to ApiResource properties

### PR DESCRIPTION
**Pull request by edamov**
| Q             | A
| ------------- | ---
| Bug fix?      | no 
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no
| Tickets       | ~fixes~ https://github.com/api-platform/docs/issues/868 
| License       | MIT
| Doc PR        | api-platform/docs#...

I started work on https://github.com/api-platform/docs/issues/868 because it was marked as `good first issue` :)
First, I added links to documentation for API Resource properties and sort properties alphabetically.

But there are some points I don't understand from this issue.
What should we do with `@see https://github.com/Haehnchen/idea-php-annotation-plugin/issues/112` in the class ApiResource?
I found this [PR](https://github.com/api-platform/core/pull/1788#pullrequestreview-107255243) but still can't understand.

### CODE REVIEW

**Comment by teohhanhui** [comment]
> What should we do with @see https://github.com/Haehnchen/idea-php-annotation-plugin/issues/112 in the class ApiResource?

Nothing. :smile:

**Comment by teohhanhui** [comment]
Hmm... Actually, this does not address https://github.com/api-platform/docs/issues/868, but I think this is something worth doing as well. WDYT @dunglas?

File: src/Annotation/ApiResource.php [hunk 1]
```diff
@@ -112,67 +109,81 @@ final class ApiResource
...
+        'securityPostDenormalize',
+        'securityPostDenormalizeMessage',
         'sunset',
         'swaggerContext',
         'validationGroups',
     ];
 
     /**
-     * @var string
+     * @see https://api-platform.com/docs/core/operations/#operations
```
Comment (teohhanhui) [inline src/Annotation/ApiResource.php]: Can we remove all the pointless hashes that are just top of the page anyway? :smile:

i.e. it should be https://api-platform.com/docs/core/operations/

File: src/Annotation/ApiResource.php [hunk 2]
```diff
@@ -112,67 +109,81 @@ final class ApiResource
...
-    public $shortName;
+    public $collectionOperations;
 
     /**
      * @var string
      */
     public $description;
 
+    /**
+     * @see https://api-platform.com/docs/core/graphql/#enabling-graphql
```
Comment (teohhanhui) [inline src/Annotation/ApiResource.php]: I think this should just point to https://api-platform.com/docs/core/graphql/

File: src/Annotation/ApiResource.php [hunk 3]
```diff
@@ -251,19 +270,38 @@ final class ApiResource
...
+     * @see https://github.com/Haehnchen/idea-php-annotation-plugin/issues/112
+     * @see https://api-platform.com/docs/core/mercure/#pushing-the-api-updates
+     *
      * @var mixed
      */
     private $mercure;
 
     /**
      * @see https://github.com/Haehnchen/idea-php-annotation-plugin/issues/112
+     * @see https://api-platform.com/docs/core/messenger/#using-messenger-with-an-input-object
```
Comment (teohhanhui) [inline src/Annotation/ApiResource.php]: Should be: https://api-platform.com/docs/core/messenger/#dispatching-a-resource-through-the-message-bus

File: src/Annotation/ApiResource.php [hunk 4]
```diff
@@ -274,128 +312,127 @@ final class ApiResource
...
+     * @var array
+     */
     private $order;
 
     /**
      * @see https://github.com/Haehnchen/idea-php-annotation-plugin/issues/112
      *
+     * @var string|false
+     */
+    private $output;
```
Comment (teohhanhui) [inline src/Annotation/ApiResource.php]: This one is https://api-platform.com/docs/core/dto/#specifying-an-input-or-an-output-data-representation

File: src/Annotation/ApiResource.php [hunk 5]
```diff
@@ -251,19 +270,38 @@ final class ApiResource
     /**
      * @see https://github.com/Haehnchen/idea-php-annotation-plugin/issues/112
      *
+     * @var string|false
+     */
+    private $input;
```
Comment (teohhanhui) [inline src/Annotation/ApiResource.php]: This one is https://api-platform.com/docs/core/dto/#specifying-an-input-or-an-output-data-representation

File: src/Annotation/ApiResource.php [hunk 6]
```diff
@@ -274,128 +312,127 @@ final class ApiResource
      *
      * @var array
      */
+    private $openapiContext;
```
Comment (teohhanhui) [inline src/Annotation/ApiResource.php]: This one is https://api-platform.com/docs/core/swagger/#using-the-openapi-and-swagger-contexts

File: src/Annotation/ApiResource.php [hunk 7]
```diff
@@ -274,128 +312,127 @@ final class ApiResource
...
      * @see https://github.com/Haehnchen/idea-php-annotation-plugin/issues/112
      *
+     * @var string|false
+     */
+    private $output;
+
+
+    /**
+     * @see https://github.com/Haehnchen/idea-php-annotation-plugin/issues/112
+     * @see https://api-platform.com/docs/core/pagination/#globally-1
```
Comment (teohhanhui) [inline src/Annotation/ApiResource.php]: Wrong link, it should be https://api-platform.com/docs/core/pagination/#for-a-specific-resource-1

File: src/Annotation/ApiResource.php [hunk 8]
```diff
@@ -274,128 +312,127 @@ final class ApiResource
...
+     *
      * @var bool
      */
     private $paginationClientEnabled;
 
     /**
      * @see https://github.com/Haehnchen/idea-php-annotation-plugin/issues/112
+     * @see https://api-platform.com/docs/core/pagination/#globally-3
```
Comment (teohhanhui) [inline src/Annotation/ApiResource.php]: Wrong link, it should be https://api-platform.com/docs/core/pagination/#for-a-specific-resource-3

**Comment by teohhanhui** [comment]
Thanks for squashing! :smile:

Can you target `2.5` instead?

**Comment by edamov** [comment]
> Thanks for squashing! 
> 
> Can you target `2.5` instead?
> 
> ```
> git reset --hard origin/2.5
> git cherry-pick 254074a
> ```

oops. I do something in a wrong way

**Comment by teohhanhui** [comment]
I can take care of it if you'd like. :smile:

**Comment by edamov** [comment]
I need to try :)

**Comment by edamov** [comment]
It got only worse :face_with_head_bandage: Now it has conflicts

**Comment by teohhanhui** [comment]
Did you try following the steps I've provided? :smile:

**Comment by edamov** [comment]
Yes. But I had no success :( I had for some reasons second master branch

**Comment by teohhanhui** [comment]
Okay, no worries. I'll take care of this.

**Comment by edamov** [comment]
Thanks a lot for your help and your time!

**Comment by teohhanhui** [comment]
Note for next time: Always use a topic branch (e.g. `feature/api-resource-annotation-link-docs`) when creating a PR. Do not use the `master` branch (or others like `2.5`).

**Comment by teohhanhui** [comment]
Thanks @edamov! :tada:

**Pull request merged**